src/main/java/Deadline.java

```diff
@@ -0,0 +1,23 @@
+import java.time.LocalDate;
+
+class Deadline extends Task {
+    private LocalDate deadline;
+    public Deadline(String name, String start) {
+        super(name);
+        this.deadline = LocalDate.parse(start);
+    }
+
+    @Override
+    public String getFileDesc() {
+        return this.isDone
+        ? "D|1|" + this.name + "|" + convertFileDate(this.deadline)
+        : "D|0|" + this.name + "|" + convertFileDate(this.deadline);
+    }
+
+    @Override
+    public String toString() {
+        return this.isDone
+                ? "[D][X] " + this.name + " (by: " + getDate(this.deadline) + ")"
+                : "[D][ ] " + this.name + " (by: " + getDate(this.deadline) + ")";
+    }
+}
```

Comment:
Perhaps, instead of leaving the isDone logic in this toString() function, you could leave it in the Task class instead so that you dont have to retype so much code if you need to create a new Task type in the future.